packages/services/knowledge/src/knowledge-event-bridge.ts

```
@@ -0,0 +1,284 @@
+      return;
+    }
+
+    try {
+      // Stop heartbeat
+      if (this.heartbeatInterval) {
+        clearInterval(this.heartbeatInterval);
+        this.heartbeatInterval = null;
+      }
+
+      // Remove event listeners
+      this.eventListeners.forEach((listener, event) => {
```

ESLint

[error] Use `for…of` instead of `.forEach(…)`.

(unicorn/no-array-for-each)